layers/newrelic/Dockerfile

```diff
@@ -0,0 +1,43 @@
+ARG PHP_VERSION
+ARG NEW_RELIC_LICENSE_KEY
+ARG NEW_RELIC_DAEMON_ADDRESS
+FROM bref/build-php-$PHP_VERSION AS ext
+
+# Build the New Relic Agent install silently for PHP Agent 9.14
+# Then append php.ini with key information including a license key and 
+# location of the daemon coming from an ENV VAR for Lambda
+
+RUN \
+  curl -L http://download.newrelic.com/php_agent/release/newrelic-php5-9.14.0.290-linux.tar.gz | tar -C /tmp -zx && \
+  export NR_INSTALL_USE_CP_NOT_LN=1 && \
+  export NR_INSTALL_SILENT=1 && \
+  /tmp/newrelic-php5-*/newrelic-install install && \
+  rm -rf /tmp/newrelic-php5-* /tmp/nrinstall* 
+  
+RUN echo $' \n \
+extension=newrelic.so \n \
+newrelic.appname = ${NEW_RELIC_APP_NAME} \n \
+newrelic.license = ${NEW_RELIC_LICENSE_KEY} \n \
+newrelic.daemon.address = ${NEW_RELIC_DAEMON_ADDRESS} \n \
+newrelic.logfile = ${NEW_RELIC_LOG_FILE} \n \
+newrelic.loglevel = ${NEW_RELIC_LOG_LEVEL} \n \
+newrelic.daemon.app_connect_timeout = ${NEW_RELIC_DAEMON_APP_CONNECT_TIMEOUT} \n \
+newrelic.daemon.start_timeout = ${NEW_RELIC_DAEMON_START_TIMEOUT} \n \
+' >> /opt/bref/etc/php/php.ini
+
+# newrelic.so will be installed to `php-config --extension-dir`/newrelic.so
+# In this core image newrelic.ini will be placed but not exist in the scatch layer
+# ${INSTALL_DIR}/etc/php/php.ini /tmp/php.ini - php.ini will contain the needed config and extension ref
+
+# Copy the newrelic.so and php.ini file to the tmp folder
+
+RUN cp `php-config --extension-dir`/newrelic.so /tmp/newrelic.so
+RUN cp /opt/bref/etc/php/php.ini /tmp/php.ini
+
+# Build the final image from the scratch image that contain files you want to export
+FROM scratch
+
+# Copy the two key files to the correct location for the empty layer.
+# newrelic.so should become a layered extension. php.ini should overwrite base layer
+COPY --from=ext /tmp/newrelic.so /opt/bref/lib/php/extensions/no-debug-zts-20190902/newrelic.so
+COPY --from=ext /tmp/php.ini /opt/bref/etc/php/php.ini
```

Comment:
```suggestion
COPY --from=ext /tmp/ext.ini /opt/bref/etc/php/conf.d/ext-newrelic.ini

```

Reply:
I can't find reference of this ext.ini and where it comes from or is used. Can you provide context or a docs link?

Reply:
Have a look at the Ldap layer: https://github.com/brefphp/extra-php-extensions/blob/0.8.1/layers/ldap/Dockerfile#L14-L22

Line 21: Write the file `/opt/bref-extra/ldap.so`
Line 15: Add `extension=/opt/bref-extra/ldap.so` to a temp file `/tmp/ext.ini`
Line 22: Move the temp file to a path that PHP reads config from

Reply:
So the New Relic installation creates a newrelic.ini in the Scan Dir. However the Scan Dir isn't being read. Which is why I was appending to php.ini for simplicity. I'll create a newrelic.ini in the same dir as php.ini as it seems from your suggestion this should work too.